src/pages/scheduler/components/Toolbar.tsx

```diff
@@ -7,6 +7,69 @@ import { logEvent } from 'lib/utils/logEvent';
 
 import { ShareButton } from './share/ShareButton';
 
+export const MobileToolBar =
+  (onSelectedDateChange: (date: Date) => void, term: string, vCalendar?: string) =>
+  ({ label, date }: ToolbarProps) => {
+    const handleClick = (offset?: number) => () => {
+      if (offset) {
+        const d = new Date(date);
+        d.setDate(d.getDate() + offset);
+        onSelectedDateChange(d);
+      } else {
+        onSelectedDateChange(new Date());
+      }
+    };
+
+    console.log(date);
+
+    const handleDownload = () => {
+      if (vCalendar) {
+        logEvent('calendar_download', { term });
+
+        const element = document.createElement('a');
+        const file = new Blob([vCalendar], { type: 'text/plain' });
+        element.href = URL.createObjectURL(file);
+        element.download = `${term}_calendar.ics`;
+        element.click();
+        document.body.removeChild(element);
+      }
+    };
+
+    return (
+      <Flex pb="0.5em" justifyContent="space-between" alignItems="center">
+        <Text fontSize="xs">{label}</Text>
+        <HStack pb="0.2em">
+          <ShareButton term={term as Term} disabled={!vCalendar} />
+          <IconButton
+            icon={<DownloadIcon />}
+            aria-label="Download timetable"
+            size="sm"
+            colorScheme="blue"
+            onClick={handleDownload}
+            disabled={!vCalendar}
+          />
+          <Button size="sm" colorScheme="gray" onClick={handleClick()}>
+            Today
+          </Button>
+          <IconButton
+            aria-label="Previous day"
+            colorScheme="gray"
+            icon={<ChevronLeftIcon />}
+            size="sm"
+            onClick={handleClick(-1)}
+          />
+          <IconButton
+            aria-label="Next day"
+            colorScheme="gray"
+            icon={<ChevronRightIcon />}
+            size="sm"
+            onClick={handleClick(1)}
+          />
+        </HStack>
+      </Flex>
+    );
+  };
```

Comment:
Do they not share enough code to make it one component? I feel this adds a lot of duplicated code.

Reply:
yaaaa I got lazy